File: src/main/java/dev/mazurkiewicz/florystyka/jwt/JwtProperties.java
package dev.mazurkiewicz.florystyka.jwt;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.http.HttpHeaders;

@Getter
@Setter
@ConfigurationProperties(prefix = "application.jwt")
public class JwtProperties {

    private String secretKey;
    private String tokenPrefix;
    private Integer tokenExpirationAfterSeconds;
    private String issuer;

    public String getAuthorizationHeader() {
        return HttpHeaders.AUTHORIZATION;
    }
}

File: src/main/kotlin/io/mazurkiewicz/quizer/auth/config/JwtProperties.kt
package io.mazurkiewicz.quizer.auth.config

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "auth.jwt")
data class JwtProperties(
    val secretKey: String,
    val expirationInSeconds: Long,
    val refreshExpirationInSeconds: Long,
    val tokenAlgorithm: String,
    val keyAlgorithm: String
)